modules/videoio/src/cap_openni2.cpp

```diff
@@ -286,15 +329,27 @@ openni::Status CvCapture_OpenNI2::toggleStream(int stream, bool toggle)
         status = streams[stream].create(device, stream_sensor_types[stream]);
         if (status == openni::STATUS_OK)
         {
-            // set video mode
-            status = streams[stream].setVideoMode(defaultStreamOutputMode(stream)); // xn::DepthGenerator supports VGA only! (Jan 2011)
-            if (status != openni::STATUS_OK)
+            const openni::Array<openni::VideoMode>& vm = streams[stream].getSensorInfo().getSupportedVideoModes();
+            openni::VideoMode dm = defaultStreamOutputMode(stream);
+            for (int i = 0; i < vm.getSize(); i++)
             {
-                CV_Error(CV_StsError, cv::format("CvCapture_OpenNI2::CvCapture_OpenNI2 : Couldn't set %s stream output mode: %s\n",
-                                                 stream_names[stream],
-                                                 openni::OpenNI::getExtendedError()));
-                streams[stream].destroy();
-                return status;
+                if (vm[i].getPixelFormat() == dm.getPixelFormat() &&
+                    vm[i].getResolutionX() == dm.getResolutionX() &&
+                    vm[i].getResolutionY() == dm.getResolutionY() &&
+                    vm[i].getFps() == dm.getFps())
+                {
+                    // try to set default mode if available
+                    // by Jan 2011 xn::DepthGenerator supported VGA only
```

Comment:
Does this comment still actual?

Reply:
I don't know about Xn and PrimeSense devices but OpenNI itself and its Kinect 2 drivers support a lot of resolutions
This comment is removed